include/clang/AST/ASTImporter.h

```diff
@@ -56,12 +57,19 @@ class TypeSourceInfo;
   /// Imports selected nodes from one AST context into another context,
   /// merging AST nodes where appropriate.
   class ASTImporter {
+    friend class ASTNodeImporter;
```

Comment:
This friend is not necessary. The `ASTNodeImporter` is already a "private" class and the needed things (for ASTImporter) can be public. (And without friend there is no reference to ASTNodeImporter from any other file.)

Reply:
AFAIK nested classes are not automatically friends (unfortunately). Also I have made `FindDeclsInToCtx` private in `ASTImporter` so that could be used only inside `ASTNodeImporter`, since that is an implementation detail.